### ISSUE TITLE
rrule with YEARLY frequency starting on a leap day

### ISSUE THREAD

**Issue by MartinThoma** [dateutil/dateutil#823]
Suppose I wanted to get when to celebrate birthdays with rrule. Then the frequency YEARLY works fine, except for leap days. There it is in fact only every 4 years.

Is there any way to deal with it directly with rrule?

    from datetime import datetime
    from dateutil.rrule import rrule, YEARLY

    n = 1
    print(list(rrule(freq=YEARLY, count=n + 1, dtstart=datetime(1990, 4, 28))))
    print(list(rrule(freq=YEARLY, count=n + 1, dtstart=datetime(1992, 2, 29))))

gives

    [datetime.datetime(1990, 4, 28, 0, 0), datetime.datetime(1991, 4, 28, 0, 0)]
    [datetime.datetime(1992, 2, 29, 0, 0), datetime.datetime(1996, 2, 29, 0, 0)]

The fact that leap days are not even mentioned [in the docs](https://dateutil.readthedocs.io/en/stable/rrule.html) makes me wonder if this might simply be a bug.

## byyearday

This might help, but only for the 28th of February:

    from datetime import datetime
    from dateutil.rrule import rrule, YEARLY

    n = 5

    bday = datetime(1990, 4, 28)
    print(list(rrule(freq=YEARLY,
                     byyearday=bday.timetuple().tm_yday,
                     count=n + 1,
                     dtstart=bday)))

    bday = datetime(1992, 2, 29)
    print(list(rrule(freq=YEARLY,
                     byyearday=bday.timetuple().tm_yday,
                     count=n + 1,
                     dtstart=bday)))

gives

    [datetime.datetime(1990, 4, 28, 0, 0), datetime.datetime(1991, 4, 28, 0, 0), datetime.datetime(1992, 4, 27, 0, 0), datetime.datetime(1993, 4, 28, 0, 0), datetime.datetime(1994, 4, 28, 0, 0), datetime.datetime(1995, 4, 28, 0, 0)]
    [datetime.datetime(1992, 2, 29, 0, 0), datetime.datetime(1993, 3, 1, 0, 0), datetime.datetime(1994, 3, 1, 0, 0), datetime.datetime(1995, 3, 1, 0, 0), datetime.datetime(1996, 2, 29, 0, 0), datetime.datetime(1997, 3, 1, 0, 0)]

**Comment by MartinThoma** [comment]
Question was also asked on SO: https://stackoverflow.com/q/52630269/562769

**Comment by MartinThoma** [comment]
Going through https://tools.ietf.org/html/rfc5545 I cannot see how YEARLY and leap days are supposed to work.

**Comment by pganssle** [comment]
Duplicate of #149. Will be fixed as part of #285, and specifically when #522 is merged. The original author of #522 seems to have mostly abandoned it, so if anyone wants to take that and run with it, please do.

**Comment by pganssle** [comment]
@MartinThoma They are supposed to work like this:

```
DTSTART:19920229T000000
RRULE:FREQ=YEARLY;COUNT=4;RSCALE=GREGORIAN;SKIP=BACKWARD"
```

If you want to fall back to the day before. Replace `SKIP=FORWARD` to fall foward, and `SKIP=OMIT` to get the current/default behavior.
